storage/src/storage/sqlite/sync.rs

```diff
@@ -0,0 +1,923 @@
+        )?;
+        let digests = stmt
+            .query_map([(limit.map(|x| x as u32)).unwrap_or(u32::MAX)], |row| row.get(0))?
+            .collect::<rusqlite::Result<Vec<Digest>>>()?;
+        Ok(digests)
+    }
+
+    fn get_record(&mut self, commitment: &Digest) -> Result<Option<SerialRecord>> {
+        self.conn
+            .query_row(
+                r"
+        SELECT
+            owner,
+            is_dummy,
+            value,
+            payload,
+            birth_program_id,
+            death_program_id,
+            serial_number_nonce,
+            commitment,
+            commitment_randomness
+        FROM miner_records
+        WHERE commitment = ?
+        ",
+                [commitment],
+                |row| {
+                    Ok(SerialRecord {
+                        owner: row
+                            .get::<_, String>(0)?
+                            .parse()
+                            .map_err(|_| rusqlite::Error::InvalidQuery)?,
+                        is_dummy: row.get(1)?,
+                        value: AleoAmount(row.get(2)?),
+                        payload: row.get(3)?,
+                        birth_program_id: row.get(4)?,
+                        death_program_id: row.get(5)?,
+                        serial_number_nonce: row.get(6)?,
+                        commitment: row.get(7)?,
+                        commitment_randomness: row.get(8)?,
+                        serial_number_nonce_randomness: None,
+                        position: None,
+                    })
+                },
+            )
+            .optional()
+            .map_err(Into::into)
+    }
+
+    fn store_records(&mut self, records: &[SerialRecord]) -> Result<()> {
+        let mut stmt = self.conn.prepare_cached(
+            r"
+        INSERT INTO miner_records (
+            owner,
+            is_dummy,
+            value,
+            payload,
+            birth_program_id,
+            death_program_id,
+            serial_number_nonce,
+            commitment,
+            commitment_randomness
+        ) VALUES (
+            ?, ?, ?, ?, ?, ?, ?, ?, ?
+        )
+        ",
+        )?;
+        for record in records {
+            stmt.execute(params![
+                record.owner.to_string(),
+                record.is_dummy,
+                record.value.0,
+                &record.payload,
+                &record.birth_program_id,
+                &record.death_program_id,
+                &record.serial_number_nonce,
+                &record.commitment,
+                &record.commitment_randomness,
+            ])?;
+        }
+        Ok(())
+    }
+
+    fn get_commitments(&mut self) -> Result<Vec<Digest>> {
+        let mut stmt = self.conn.prepare_cached(
+            r"
+        SELECT
+        transactions.new_commitment1,
+        transactions.new_commitment2
+        FROM transactions
+        INNER JOIN transaction_blocks ON transaction_blocks.transaction_id = transactions.id
```

Comment:
do we even still need a `transaction_blocks` table now? couldn't we just store `block_hash` and `block_position` in `transactions`, together with all the other transaction info?

Reply:
I don't think we even need to store `block_position` now that we no longer store transactions as per-block blobs containing multiple of them